### PR TITLE
[excel] (Parameters) Add information about in-client parameterized scripts

### PR DESCRIPTION
**Pull request by AlexJerabek**
This PR documents the upcoming feature of user-prompts for parameterized scripts. It works the existing documentation for Power Automate parameters and returns to account for this shared information.

### CODE REVIEW

**Comment by AlexJerabek** [comment]
@ElizabethSamuel-MSFT, any suggestions for how to make that image render smaller?

**Comment by ElizabethSamuel-MSFT** [comment]
> @ElizabethSamuel-MSFT, any suggestions for how to make that image render smaller?

@AlexJerabek You may need to [actually resize it](https://review.learn.microsoft.com/en-us/help/contribute/contribute-how-to-format-screenshot?branch=main#resize-images).

File: docs/develop/user-input.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ * @param highlightThreshold The value used for comparisons.
+ * @param color A string representing the color to make the high value cells. 
+ *   This must be a color code representing the color of the background, 
+ *   in the form #RRGGBB (e.g., "FFA500") or a named HTML color (e.g., "orange").
+ */
+function main(
+  workbook: ExcelScript.Workbook, 
+  highlightThreshold: number, 
+  color: string) {
+    // Get the used cells in th1e current worksheet.
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/user-input.md:33]:
```suggestion
    // Get the used cells in the current worksheet.
```

File: docs/develop/user-input.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+}
+```
+
+## `main` parameters: Pass data to a script
+
+All script input is specified as additional parameters for the `main` function. New parameters are added after the mandatory `workbook: ExcelScript.Workbook` parameter. For example, if you wanted a script to accept a `string` that represents a name as input, you would change the `main` signature to `function main(workbook: ExcelScript.Workbook, name: string)`.
+
+### Optional parameters
+
+Optional parameters don't need the user to provide a value. This implies your script either has default behavior or this parameter is only needed in a corner-case. They are denoted in your script with the [optional modifier](https://www.typescriptlang.org/docs/handbook/2/functions.html#optional-parameters) `?`. For example, in `function main(workbook: ExcelScript.Workbook, Name?: string)` the parameter `Name` is optional.
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/user-input.md:54]:
```suggestion
Optional parameters don't need the user to provide a value. This implies your script either has default behavior or this parameter is only needed in a corner case. They're denoted in your script with the [optional modifier](https://www.typescriptlang.org/docs/handbook/2/functions.html#optional-parameters) `?`. For example, in `function main(workbook: ExcelScript.Workbook, Name?: string)` the parameter `Name` is optional.
```

File: docs/develop/user-input.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+### Default parameter values
+
+[Default parameter values](https://www.typescriptlang.org/docs/handbook/variable-declarations.html#default-values) automatically fill the action's field with a value. To set a default value, assign a value to the parameter in the `main` signature. For example, in `function main(workbook: ExcelScript.Workbook, location: string = "Seattle")` the parameter `location` has the value `"Seattle"` unless something else is provided.
+
+### Drop-down lists for parameters
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/user-input.md:60]:
```suggestion
### Dropdown lists for parameters
```

File: docs/develop/user-input.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+Help others using your script in their flow by providing a list of acceptable parameter choices. If there is a small subset of values that your script uses, create a parameter that is those literal values. Do this by declaring the parameter type to be a [union of literal values](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types). For example, in `function main(workbook: ExcelScript.Workbook, location: "Seattle" | "Redmond")` the parameter `location` can only be `"Seattle"` or `"Redmond"`. When the script is run, users get a drop-down list with those two options.
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/user-input.md:62]:
```suggestion
Help others using your script in their flow by providing a list of acceptable parameter choices. If there's a small subset of values that your script uses, create a parameter that is those literal values. Do this by declaring the parameter type to be a [union of literal values](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types). For example, in `function main(workbook: ExcelScript.Workbook, location: "Seattle" | "Redmond")` the parameter `location` can only be `"Seattle"` or `"Redmond"`. When the script is run, users get a dropdown list with those two options.
```

File: docs/develop/user-input.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+When adding input parameters and return values, consider the following allowances and restrictions.
+
+1. The first parameter must be of type `ExcelScript.Workbook`. Its parameter name doesn't matter.
+
+1. The types `string`, `number`, `boolean`, `unknown`, and `object`.
+
+1. Arrays (both `[]` and `Array<T>` styles) of the previously listed types are supported. Nested arrays are also supported.
+
+1. Union types are allowed if they are a union of literals belonging to a single type (such as `"Left" | "Right"`, not `"Left" | 5`).
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/user-input.md:89]:
```suggestion
1. Union types are allowed if they're a union of literals belonging to a single type (such as `"Left" | "Right"`, not `"Left" | 5`).
```

File: docs/develop/user-input.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+description: Add parameters to Office Scripts so users can control their experience. 
+ms.date: 08/15/2023
+ms.localizationpriority: medium
+---
+
+# Get user input for scripts
+
+Adding parameters to your script lets other users provide data for the script, without needing to edit code. When your script is run through the ribbon or a button, a prompt pops up that asks for input.
+
+:::image type="content" source="../images/user-input-example.png" alt-text="The dialog box shown to users when a script with parameters is run.":::
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/user-input.md:12]: In the image, you should replace your name with some test name

Comment (AlexJerabek) [reply docs/develop/user-input.md:12]: Good catch! I removed that file location section for now (the UI might change and the "OneDrive" gets cutoff anyway)

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Pull request merged**